### ISSUE TITLE
Building desktop on manjaro linux fails with "error: failed to run custom build command for `devpod-desktop v0.1.0"

### ISSUE THREAD

**Issue by josephwinston** [loft-sh/devpod#424]
TL;DR - How do I get a real message out to debug my setup? Passing `--verbose` to yarn doesn't help

Steps:
* Clone repo
* `cd desktop`
* `../hack/rebuild.sh`
* `yarn`
* `cd src-tauri && cargo update && cd ..`

Fails:

``` $yarn tauri dev
yarn run v1.22.19
$ tauri dev
     Running BeforeDevCommand (`yarn dev`)
$ vite

  VITE v4.3.7  ready in 398 ms

  ➜  Local:   http://localhost:1420/
  ➜  Network: use --host to expose
        Info Watching /home/josephwinston/src/Remote/GIT/loft-sh/devpod/desktop/src-tauri for changes...
   Compiling proc-macro2 v1.0.60
   Compiling quote v1.0.28
   Compiling unicode-ident v1.0.9
   Compiling serde v1.0.164
   Compiling autocfg v1.1.0
   Compiling smallvec v1.10.0
   Compiling cfg-if v1.0.0
   Compiling pkg-config v0.3.27
   Compiling libc v0.2.146
   Compiling hashbrown v0.12.3
   Compiling heck v0.4.1
   Compiling winnow v0.4.7
   Compiling target-lexicon v0.12.8
   Compiling version-compare v0.1.1
   Compiling once_cell v1.18.0
   Compiling syn v1.0.109
   Compiling indexmap v1.9.3
   Compiling bitflags v1.3.2
   Compiling ppv-lite86 v0.2.17
   Compiling siphasher v0.3.10
   Compiling getrandom v0.1.16
   Compiling version_check v0.9.4
   Compiling parking_lot_core v0.9.8
   Compiling lock_api v0.4.10
   Compiling thiserror v1.0.40
   Compiling scopeguard v1.1.0
   Compiling fnv v1.0.7
   Compiling phf_shared v0.8.0
   Compiling pin-project-lite v0.2.9
   Compiling syn v2.0.18
   Compiling cfg-expr v0.15.3
   Compiling itoa v1.0.6
   Compiling futures-core v0.3.28
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling phf_shared v0.10.0
   Compiling slab v0.4.8
   Compiling futures-task v0.3.28
   Compiling getrandom v0.2.10
   Compiling anyhow v1.0.71
   Compiling rand_core v0.5.1
   Compiling rand_core v0.6.4
   Compiling futures-util v0.3.28
   Compiling rand_chacha v0.2.2
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.3.1
   Compiling pin-utils v0.1.0
   Compiling proc-macro-error-attr v1.0.4
   Compiling byteorder v1.4.3
   Compiling futures-channel v0.3.28
   Compiling semver v1.0.17
   Compiling rand v0.7.3
   Compiling rand v0.8.5
   Compiling value-bag v1.4.0
   Compiling proc-macro-error v1.0.4
   Compiling precomputed-hash v0.1.1
   Compiling log v0.4.19
   Compiling ryu v1.0.13
   Compiling cc v1.0.79
   Compiling new_debug_unreachable v1.0.4
   Compiling mac v0.1.1
   Compiling memchr v2.5.0
   Compiling futf v0.1.5
   Compiling phf_generator v0.8.0
   Compiling phf_generator v0.10.0
   Compiling parking_lot v0.12.1
   Compiling utf-8 v0.7.6
   Compiling phf_codegen v0.8.0
   Compiling string_cache_codegen v0.5.2
   Compiling unicode-segmentation v1.10.1
   Compiling dtoa v1.0.6
   Compiling tendril v0.4.3
   Compiling percent-encoding v2.3.0
   Compiling tinyvec_macros v0.1.1
   Compiling dtoa-short v0.3.4
   Compiling selectors v0.22.0
   Compiling tinyvec v1.6.0
   Compiling heck v0.3.3
   Compiling rustc_version v0.4.0
   Compiling cfg-expr v0.9.1
   Compiling nodrop v0.1.14
   Compiling markup5ever v0.10.1
   Compiling strsim v0.10.0
   Compiling matches v0.1.10
   Compiling itoa v0.4.8
   Compiling stable_deref_trait v1.2.0
   Compiling convert_case v0.4.0
   Compiling version-compare v0.0.11
   Compiling ident_case v1.0.1
   Compiling servo_arc v0.1.1
   Compiling form_urlencoded v1.2.0
   Compiling fxhash v0.2.1
   Compiling unicode-bidi v0.3.13
   Compiling thin-slice v0.1.1
   Compiling alloc-no-stdlib v2.0.4
   Compiling gio v0.15.12
   Compiling darling_core v0.20.1
   Compiling same-file v1.0.6
   Compiling alloc-stdlib v0.2.2
   Compiling unicode-normalization v0.1.22
   Compiling brotli-decompressor v2.3.4
   Compiling walkdir v2.3.3
   Compiling futures-io v0.3.28
   Compiling crc32fast v1.3.2
   Compiling serde_json v1.0.97
   Compiling memoffset v0.9.0
   Compiling idna v0.4.0
   Compiling glob v0.3.1
   Compiling dunce v1.0.4
   Compiling simd-adler32 v0.3.5
   Compiling openssl-sys v0.9.90
   Compiling field-offset v0.3.6
   Compiling adler v1.0.2
   Compiling typenum v1.16.0
   Compiling brotli v3.3.4
   Compiling miniz_oxide v0.7.1
   Compiling uuid v1.3.4
   Compiling generic-array v0.14.7
   Compiling gtk v0.15.5
   Compiling tokio v1.28.2
   Compiling flate2 v1.0.26
   Compiling cfb v0.7.3
   Compiling fdeflate v0.3.0
   Compiling num_cpus v1.15.0
   Compiling socket2 v0.4.9
   Compiling mio v0.8.8
   Compiling png v0.17.9
   Compiling x11 v2.21.0
   Compiling infer v0.12.0
   Compiling tracing-core v0.1.31
   Compiling openssl v0.10.55
   Compiling foreign-types-shared v0.1.1
   Compiling crossbeam-utils v0.8.16
   Compiling foreign-types v0.3.2
   Compiling serde_derive v1.0.164
   Compiling thiserror-impl v1.0.40
   Compiling futures-macro v0.3.28
   Compiling cssparser-macros v0.6.1
   Compiling darling_macro v0.20.1
   Compiling phf_macros v0.8.0
   Compiling cssparser v0.27.2
   Compiling html5ever v0.25.2
   Compiling derive_more v0.99.17
   Compiling darling v0.20.1
   Compiling phf_macros v0.10.0
   Compiling serde_with_macros v3.0.0
   Compiling ctor v0.1.26
   Compiling tracing v0.1.37
   Compiling phf v0.8.0
   Compiling openssl-macros v0.1.1
   Compiling aho-corasick v1.0.2
   Compiling x11-dl v2.21.0
   Compiling libloading v0.7.4
   Compiling lazy_static v1.4.0
   Compiling phf v0.10.1
   Compiling regex-syntax v0.7.2
   Compiling raw-window-handle v0.5.2
   Compiling httparse v1.8.0
   Compiling futures-sink v0.3.28
   Compiling native-tls v0.2.11
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling futures-executor v0.3.28
   Compiling tao v0.16.2
   Compiling dirs-sys-next v0.1.2
   Compiling try-lock v0.2.4
   Compiling io-lifetimes v1.0.11
   Compiling openssl-probe v0.1.5
   Compiling dirs-next v2.0.0
   Compiling want v0.3.1
   Compiling digest v0.10.7
   Compiling crossbeam-channel v0.5.8
   Compiling regex v1.8.4
   Compiling instant v0.1.12
   Compiling httpdate v1.0.2
   Compiling wry v0.24.3
   Compiling cpufeatures v0.2.8
   Compiling base64 v0.21.2
   Compiling rustix v0.37.20
   Compiling tower-service v0.3.2
   Compiling tauri-runtime v0.14.0
   Compiling sha2 v0.10.7
   Compiling ico v0.3.0
   Compiling aho-corasick v0.7.20
   Compiling bstr v1.5.0
   Compiling tempfile v3.6.0
   Compiling tauri-runtime-wry v0.14.0
   Compiling linux-raw-sys v0.3.8
   Compiling rfd v0.10.0
   Compiling http-range v0.1.5
   Compiling globset v0.4.10
   Compiling serialize-to-javascript-impl v0.1.1
   Compiling filetime v0.2.21
   Compiling xattr v0.2.3
   Compiling tauri v1.4.1
   Compiling thread_local v1.1.7
   Compiling encoding_rs v0.8.32
   Compiling fastrand v1.9.0
   Compiling ipnet v2.7.2
   Compiling time-core v0.1.1
   Compiling mime v0.3.17
   Compiling num_threads v0.1.6
   Compiling pathdiff v0.2.1
   Compiling time v0.3.22
   Compiling open v3.2.0
   Compiling ignore v0.4.20
   Compiling tar v0.4.38
   Compiling serde_repr v0.1.12
   Compiling zip v0.6.6
   Compiling serde_spanned v0.6.2
   Compiling toml_datetime v0.6.2
   Compiling toml v0.5.11
   Compiling url v2.4.0
   Compiling toml_edit v0.19.10
   Compiling bytes v1.4.0
   Compiling serde_with v3.0.0
   Compiling string_cache v0.8.7
   Compiling system-deps v5.0.0
   Compiling http v0.2.9
   Compiling treediff v4.0.2
   Compiling toml v0.7.4
   Compiling proc-macro-crate v1.3.1
   Compiling soup2-sys v0.2.0
   Compiling javascriptcore-rs-sys v0.4.0
   Compiling system-deps v6.1.0
   Compiling json-patch v1.0.0
   Compiling http-body v0.4.5
   Compiling embed-resource v2.1.1
   Compiling serde_urlencoded v0.7.1
   Compiling cargo_toml v0.15.3
   Compiling tauri-winres v0.1.1
   Compiling serialize-to-javascript v0.1.1
   Compiling os_info v3.7.0
   Compiling shared_child v1.0.0
   Compiling kuchiki v0.8.1
   Compiling os_pipe v1.1.4
   Compiling vte_generate_state_changes v0.1.1
   Compiling num-traits v0.2.15
   Compiling arrayvec v0.5.2
   Compiling state v0.5.3
   Compiling option-ext v0.2.0
   Compiling sys-locale v0.2.4
   Compiling utf8parse v0.2.1
   Compiling tauri-utils v1.4.0
   Compiling minisign-verify v0.2.1
   Compiling dirs-sys v0.4.1
   Compiling utf8-width v0.1.6
   Compiling byte-unit v4.0.19
   Compiling vte v0.10.1
   Compiling glib-sys v0.15.10
   Compiling gobject-sys v0.15.10
   Compiling gdk-sys v0.15.1
   Compiling gio-sys v0.15.10
   Compiling atk-sys v0.15.1
   Compiling pango-sys v0.15.10
   Compiling cairo-sys-rs v0.15.1
   Compiling gdk-pixbuf-sys v0.15.10
   Compiling gtk-sys v0.15.3
   Compiling glib-macros v0.15.13
   Compiling gtk3-macros v0.15.6
   Compiling webkit2gtk-sys v0.18.0
   Compiling gdkx11-sys v0.15.1
   Compiling strip-ansi-escapes v0.1.1
   Compiling tokio-util v0.7.8
   Compiling tokio-native-tls v0.3.1
   Compiling dirs v5.0.1
   Compiling fern v0.6.2
   Compiling time v0.1.45
   Compiling iana-time-zone v0.1.57
   Compiling fix-path-env v0.0.0 (https://github.com/tauri-apps/fix-path-env-rs#6c63c49e)
   Compiling serde_qs v0.12.0
   Compiling h2 v0.3.19
   Compiling chrono v0.4.26
   Compiling tauri-codegen v1.4.0
   Compiling tauri-build v1.4.0
   Compiling gdkwayland-sys v0.15.3
   Compiling tauri-plugin-deep-link v0.1.1
   Compiling glib v0.15.12
   Compiling libappindicator-sys v0.7.3
   Compiling hyper v0.14.26
   Compiling tauri-macros v1.4.0
   Compiling devpod-desktop v0.1.0 (/home/josephwinston/src/Remote/GIT/loft-sh/devpod/desktop/src-tauri)
error: failed to run custom build command for `devpod-desktop v0.1.0 (/home/josephwinston/src/Remote/GIT/loft-sh/devpod/desktop/src-tauri)`

Caused by:
  process didn't exit successfully: `/home/josephwinston/src/Remote/GIT/loft-sh/devpod/desktop/src-tauri/target/debug/build/devpod-desktop-8a290a6943c41741/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=TAURI_CONFIG
  cargo:rerun-if-changed=tauri.conf.json
  cargo:rustc-cfg=desktop
  cargo:rustc-cfg=dev
  path matching bin/devpod-cli-x86_64-unknown-linux-gnu not found.
warning: build failed, waiting for other jobs to finish...

🚀 A new version of Tauri CLI is available! [2.0.0-alpha.10]
error Command failed with exit code 101.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by pascalbreuninger** [comment]
Hi, looks like it's missing the cli binary, have you run ./hack/rebuild.sh before the build?

**Comment by josephwinston** [comment]
Progress!

I attempted to simply run 
```
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags "-s -w" -o test/devpod-cli-linux-amd64
```
from `./hack/rebuild.sh` and received strange errors.

I then noticed that I had an obsolete version of `go` on my path and updated to `go1.20.5`

Now `./hack/rebuild.sh` runs without errors as does `yarn tauri dev`.

I'm now seeing the following error that I also see in the [Linux AppImage](https://github.com/loft-sh/devpod/releases/latest/download/DevPod_linux_amd64.AppImage)

```
Error: failed to run shell echo: Unknown option: `-lc'
Usage: tcsh [ -bcdefilmnqstvVxX ] [ argument ... ].
```

Yes I now that I am a dinosaur who has `tcsh` for my default shell.  (LONG story but it has to do with dotfiles from the dark ages before bash when Sun and SGI ruled the earth.) I ran `chsh -s /bin/bash` and set the environment variable SHELL to `/bin/bash`.

Now I have real errors that I can fix.
